[CARBONDATA-3791] Fix documentation for various features (#3738)
apache/carbondata: docs/index-server.md

## Background

Carbon currently prunes and caches all block/blocklet index information into the driver for

> normal table, for Bloom/Index indexes the JDBC driver will launch a job to prune and cache the

Reviewer: ```suggestion
normal table, for Bloom/Lucene indexes the JDBC driver will launch a job to prune and cache the
```

Author: done

> datamaps in executors.

Reviewer: ```suggestion
indexes in executors.
```

Author: done

**Note:** Multiple JDBC drivers can connect to the index server to use the cache.

> ## Enabling Size based distribution for Legacy stores

Reviewer: Please correct the format in Line No.55

> The default round robin based distribution causes unequal distribution of cache amoung the executors, which can cause any 1 of the executors to be bloated with too much cache and cause performance degrade.

Reviewer: ```suggestion
The default round robin based distribution causes unequal distribution of cache among the executors, which can cause any 1 of the executors to be bloated with too much cache and cause performance degrade.
```

Author: done

> The default round robin based distribution causes unequal distribution of cache amoung the executors, which can cause any 1 of the executors to be bloated with too much cache and cause performance degrade.

Reviewer 2: ```suggestion
The default round robin based distribution causes unequal distribution of cache among the executors, which can cause any one of the executors to be bloated with too much cache resulting in performance degrade.
```

Author: done

> This problem can be solved by running the upgrade_segment command which will fill the data size values for each segment in the tablestatus file. Any cache loaded after this can use the traditional size based distribution.

Reviewer: ```suggestion
This problem can be solved by running the `upgrade_segment` command which will fill the data size values for each segment in the tablestatus file. Any cache loaded after this can use the traditional size based distribution.
```

Author: done
